dds_web/scheduled_tasks.py

```diff
@@ -22,3 +22,49 @@ def change_status_to_expired():
         # ):
 
         #     flask.current_app.logger.debug("Project: %s - Expires: %s", project, project.expires)
+
+
+@scheduler.task("cron", id="expired_to_archived", minute=1, hour=0, misfire_grace_time=1)
+def set_expired_to_archived():
+    import sqlalchemy
+    from dds_web import db
+    from dds_web.database import models
+    from dds_web.errors import DatabaseError
+    from dds_web.utils import current_time
+    from dds_web.api.project import ProjectStatus
+
+    with scheduler.app.app_context():
+        archived_projs = db.session.query(models.ProjectStatuses.project_id).filter(
+            models.ProjectStatuses.status == "Archived"
+        )
+        expired_projs = models.ProjectStatuses.query.filter(
+            models.ProjectStatuses.project_id.notin_(archived_projs),
+            models.ProjectStatuses.status == "Expired",
+            models.ProjectStatuses.deadline <= current_time(),
+        ).all()
+
+        archive = ProjectStatus()
+        for proj_status in expired_projs:
+            new_status_row, delete_message = archive.archive_project(
+                project=proj_status.project,
+                current_time=current_time(),
+            )
+            proj_status.project.project_statuses.append(new_status_row)
+            flask.current_app.logger.debug(delete_message)
+
+        try:
+            db.session.commit()
+        except (sqlalchemy.exc.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
+            flask.current_app.logger.exception(err)
+            db.session.rollback()
+            raise DatabaseError(
+                message=str(err),
+                alt_message=(
+                    "Status was not updated"
+                    + (
+                        ": Database malfunction."
+                        if isinstance(err, sqlalchemy.exc.OperationalError)
+                        else ": Server Error."
+                    )
+                ),
+            ) from err
```

Comment:
Wondering if maybe we should commit once per project.....

Reply:
If you think that works better. Would it add to the overhead? I guess it depends on how many projects there are.

Reply:
Not entirely convinced but I'm thinking it won't be many projects in one go and if there's an error in one but not the rest then none of them will be changed... So yeah I think change to commiting once per project.

Reply:
Done!